Add verification case val-2d (#204)
idaholab/TMAP8: doc/content/verification_and_validation/val-2d.md

D = D_{0} \exp \left( - \frac{E_{D}}{k_b T} \right),
\end{equation}

where $E_{D}$ is the diffusion activation energy, $k_b$ is the Boltzmann’s constant, $T$ is the temperature, and $D_{0}$ is the maximum diffusivity coefficient. TMAP7 ([!citep](ambrosek2008verification)) assigns two different value, $D_{0,l}$ and $D_{0,r}$, for implantation zone ($x < 15 \times 10 ^ {-9}$ m) and other zone ($x > 15 \times 10 ^ {-9}$ m).

$\alpha_t^i$ and $\alpha_r^i$ are defined as:

\begin{equation} \label{eq:trapping}
\alpha_t^i = \alpha_{t0}^i \exp(-\epsilon_t^i / T),
\end{equation}

and

\begin{equation} \label{eq:release}
\alpha_r^i = \alpha_{r0}^i \exp(-\epsilon_r^i / T),
\end{equation}

where $\alpha_{t0}^i$ and $\alpha_{r0}^i$ are pre-exponential factors of trapping and release rate coefficients, respectively, $\epsilon_t^i$ and $\epsilon_r^i$ are the trapping and detrapping energies, respectively.

The thermal diffusion after 5000s is governing by:

\begin{equation} \label{eq:thermal}
\rho C_P \frac{d T}{d t} = \nabla D_T \nabla T,
\end{equation}

where $\rho$ is the density of Tungsten, $C_P$ is the specific heat, $D_T$ is the thermal conductivity.

> All the model parameters are taken from [!cite](hino1998hydrogen,haasz1999effect,anderl1992deuterium,frauenfelder1969solution,ambrosek2008verification) and listed in [val-2d_set_up_values].

Reviewer: ```suggestion
All the model parameters are taken from [!cite](hino1998hydrogen,haasz1999effect,anderl1992deuterium,frauenfelder1969solution,ambrosek2008verification) and listed in [val-2d_set_up_values]. Whenever the original values are updated, it is specified in [val-2d_set_up_values]. 
```

## Model Description

> In this case, TMAP8 simulates a one-dimensional domain to represent the tritium implantation, diffusion, and trapping and detrapping in a raising thermal field. Note that this case can easily be extended to a two- or three-dimensional case.

Reviewer: ```suggestion
In this case, TMAP8 simulates a one-dimensional domain to represent the tritium implantation, diffusion, and trapping and detrapping throughout the thermal history. Note that this case can easily be extended to a two- or three-dimensional case.
```

The source term in the model is described as a normal distribution in [eq:normal_distribution]:

\begin{equation} \label{eq:normal_distribution}
S = F \frac{1}{\sigma \sqrt{2 \pi}} \exp \left( - \frac{(x - \mu )^2}{2 \sigma^2} \right),
\end{equation}

where $F$ is the implantation flux provided in [val-2d_set_up_values], $\sigma = 3 \times 10^{-9}$ m is the characteristic width of the normal distribution, and $\mu = 4.6 \times 10^{-9}$ m is the depth of the normal distribution from the upstream side.

> The pressures on the upstream and downstream sides are close to vacuum pressures, and have less impact on tritium desorption. Thus, TMAP8 ignores the pressure in these enclosures. Also, because both surfdep and Dirichlet boundary condition have similar desorption performance in the near vacuum environment, we use Dirichlet boundary condition to improve the modal efficiency. The concentrations on upstream and downstream sides are defined as:

Reviewer: ```suggestion
The pressures on the upstream and downstream sides are close to vacuum pressures, and do not have a significant impact on tritium desorption. Thus, TMAP8 ignores the pressure in these enclosures for simplification. Also, because both surfdep and Dirichlet boundary conditions have similar desorption performance in the near vacuum environment, we use Dirichlet boundary condition to improve the model efficiency. The concentrations on upstream and downstream sides are defined as:
```

C = 0.
\end{equation}

Due to the high thermal conductivity in Tungsten, the model simplifies the thermal diffusion as a instantaneous process to increase the model efficiency. The temperature inside the Tungsten sample is consistent with the temperature on surface of Tungsten sample. The temperatures is 300 K during implantation, and the temperature after implantation on the upstream and downstream sides are defined as:

\begin{equation} \label{eq:temperature}
T =
    \begin{cases}
        T_l &, t < 5000 s \\
        T_l + k_T (t - 5000) &, 5000 s  < t < 6167.6 s \\
        T_h &, t > 6167.6 s
    \end{cases}
,
\end{equation}

where $T_l = 300$ K and $T_h = 1273$ K are the lowest and highest temperatures, $k_T$ is the ramp rate of temperature.

The objective of this simulation is to determine the desorption flux on the sample. This case only considers the flux on upstream side due to the marginally filled sample during the implantation. The simulation results match the experimental data published in [!cite](hino1998hydrogen) and reproduced in [val-2d_comparison].

## Case and Model Parameters

All the model parameters are listed in [val-2d_set_up_values]:

> !table id=val-2d_set_up_values caption=Values of material properties. Note that  are currently not used in the input file

Reviewer: ```suggestion
!table id=val-2d_set_up_values caption=Values of material properties. Note that parameters marked with * are currently not used in the input file. 
```

| $\epsilon_r^2$ | release energy for trap 2       | 1.60 / k_b                                                  | K                     | [!cite](ambrosek2008verification,anderl1992deuterium) |
| $\epsilon_r^3$ | release energy for trap 3       | 3.10 / k_b                                                  | K                     | [!cite](ambrosek2008verification,frauenfelder1969solution) |
| $\chi^2$  | maximum atom fraction in trap 2      | 0.00175                                                     | -                     | Adjusted from [!cite](ambrosek2008verification) |
| $\chi^3$  | maximum atom fraction in trap 3      | 0.00200                                                     | -                     | [!cite](ambrosek2008verification) |
| $\alpha_{t0}$ | pre-factor of trapping rate coefficient | 9.1316 $\times 10^{12}$                              | 1/s                   | [!cite](ambrosek2008verification) |
| $\alpha_{r0}$ | pre-factor of release rate coefficient  | 8.4 $\times 10^{12}$                                 | 1/s                   | [!cite](ambrosek2008verification) |

> | $A$       | area of Tungsten sample              | 0.0025                                                      | m                     | [!cite](hino1998hydrogen) |

Reviewer: ```suggestion
| $A$       | * area of Tungsten sample              | 0.0025                                                      | m                     | [!cite](hino1998hydrogen) |
```

| $l$       | thickness of Tungsten sample         | 1 $\times 10^{-4}$                                          | m                     | [!cite](hino1998hydrogen) |
| $T_l$     | lowest temperature                   | 300                                                         | K                     | [!cite](hino1998hydrogen) |
| $T_h$     | highest temperature                  | 1273                                                        | K                     | [!cite](hino1998hydrogen) |
| $k_T$     | raising rate of temperature          | 50                                                          | K/min                 | [!cite](hino1998hydrogen) |

!alert note title=The $\chi^2$ is adjusted to better correspond to experimental results
TMAP7 ([!citep](ambrosek2008verification)) adjusts the maximum atom fraction as 0.0041 in trap 2 from [!cite](anderl1992deuterium) to better correspond to experimental results. Thus, we also adjust it as 0.00175 to correspond to experimental results from [!cite](hino1998hydrogen).

## Results

> In this case, there is a general background drift on desorption flux due to an increasing source of atoms going into the gas phase as the heated region spread with time. Thus, we add a ramped signal peaking at 4.87 $\times 10^{17}$ H$_2$/m$^2$/s to the results of the TMAP8 during the thermal desorption. [val-2d_comparison] shows the comparison of the TMAP8 calculation and the experimental data. There is reasonable agreement between the TMAP predictions and the experimental data with the root mean square percentage error of RMSPE = 33.31 %. Note that the agreement could be improved by adjusting the model parameters and adding more potential traps. It is difficult for TMAP7 to add more than three traps, but TMAP8 manages this without any issues. It is also possible to perform this optimization with [MOOSE's stochastic tools module](https://mooseframework.inl.gov/modules/stochastic_tools/index.html).

Reviewer: ```suggestion
In this case, there is a general background drift on desorption flux due to an increasing source of atoms going into the gas phase as the heated region spread with time. Thus, we add a ramped signal peaking at 4.87 $\times 10^{17}$ H$_2$/m$^2$/s to the results of the TMAP8 during the thermal desorption. 
[val-2d_comparison] shows the comparison of the TMAP8 calculation and the experimental data. There is reasonable agreement between the TMAP predictions and the experimental data with the root mean square percentage error of RMSPE = 33.31 %. 
Note that the agreement could be improved by adjusting the model parameters and adding more potential traps. TMAP7 is limited to three traps, but TMAP8 can introduce an arbitrarily number of trapping populations. It is also possible to perform this optimization with [MOOSE's stochastic tools module](https://mooseframework.inl.gov/modules/stochastic_tools/index.html).
```

There are several reasons for the no exact fit with the data from [!cite](hino1998hydrogen): the most prominent one is the two-dimensionality of the experiment arising from beam non-uniformity and radial diffusion [!citep](anderl1992deuterium). The actual trap energies are probably a little lower than the ones indicated above if the time lag caused by two-dimensionality is significant. Exchange of hydrogen with chamber surfaces, particularly the sample support structure, may also be a factor.

> One reason the measured signal falls off while the computed one shown does not is that the source of additional atoms in the experiment may be an expanding area that grow more or less linearly, while the sample is being heated but stopped growing and thus stop emitting when the heating stop.

Reviewer: ```suggestion
One reason the measured signal falls off after \approx 6300 s while the computed one remains steady is that the source of additional atoms in the experiment may be an expanding area that grow non-linearly, while the sample is being heated but stopped growing and thus stops emitting when the heating stops.
```

!media comparison_val-2d.py
       image_name=val-2d_comparison.png
       style=width:50%;margin-bottom:2%;margin-left:auto;margin-right:auto
       id=val-2d_comparison
       caption=Comparison of TMAP8 calculation with the experimental data on the upstream side of the sample.

## Input files

!style halign=left

> The input file for this case can be found at [/val-2d.i]. The input file is different from the input file used as test in TMAP8. To limit the computational costs of the test case, the test run a version of the file with a coarser mesh and fewer time steps. More information about the changes can be found in the test specification file for this case, namely [/val-2d/tests].

Reviewer: ```suggestion
The input file for this case can be found at [/val-2d.i]. The input file is different from the input file used as test in TMAP8. To limit the computational costs of the test case, the test runs a version of the file with a coarser mesh and fewer time steps. More information about the changes can be found in the test specification file for this case, namely [/val-2d/tests].
```

Note that the agreement could be improved by adjusting the model parameters and adding more potential traps. TMAP7 is limited to three traps, but TMAP8 can introduce an arbitrarily number of trapping populations. It is also possible to perform this optimization with [MOOSE's stochastic tools module](https://mooseframework.inl.gov/modules/stochastic_tools/index.html).

There are several reasons for the no exact fit with the data from [!cite](hino1998hydrogen): the most prominent one is the two-dimensionality of the experiment arising from beam non-uniformity and radial diffusion [!citep](anderl1992deuterium). The actual trap energies are probably a little lower than the ones indicated above if the time lag caused by two-dimensionality is significant. Exchange of hydrogen with chamber surfaces, particularly the sample support structure, may also be a factor.

One reason the measured signal falls off after \approx 6300 s while the computed one remains steady is that the source of additional atoms in the experiment may be an expanding area that grow non-linearly, while the sample is being heated but stopped growing and thus stops emitting when the heating stops.

> !media comparison_val-2d.py

Reviewer: The TMAP8 results do not look that great right now. 
It looks like the time steps are really large and the curve does not look smooth.

[val-2d_comparison] shows the comparison of the TMAP8 calculation and the experimental data. There is reasonable agreement between the TMAP predictions and the experimental data with the root mean square percentage error of RMSPE = 32.81 %.
Note that the agreement could be improved by adjusting the model parameters and adding more potential traps. TMAP7 is limited to three traps, but TMAP8 can introduce an arbitrarily number of trapping populations. It is also possible to perform this optimization with [MOOSE's stochastic tools module](https://mooseframework.inl.gov/modules/stochastic_tools/index.html).

There are several reasons for the no exact fit with the data from [!cite](hino1998hydrogen): the most prominent one is the two-dimensionality of the experiment arising from beam non-uniformity and radial diffusion [!citep](anderl1992deuterium). The actual trap energies are probably a little lower than the ones indicated above if the time lag caused by two-dimensionality is significant. Exchange of hydrogen with chamber surfaces, particularly the sample support structure, may also be a factor.

> One reason the measured signal falls off after \approx 6300 s while the computed one remains steady is that the source of additional atoms in the experiment may be an expanding area that grow non-linearly, while the sample is being heated but stopped growing and thus stops emitting when the heating stops.

Reviewer: ```suggestion
One reason the measured signal falls off after $\approx$ 6300 s while the computed one remains steady is that the source of additional atoms in the experiment may be an expanding area that grow non-linearly, while the sample is being heated but stopped growing and thus stops emitting when the heating stops.
```

## Case Description

This validation problem is taken from [!cite](hino1998hydrogen) with multiple trapping capability. This case is part of the validation suite of TMAP7 as val-2d [!citep](ambrosek2008verification).

> In this experiment, tritium is implanted at 5 keV and a flux of 1 $\times 10^{19}$ atom/m$_2$/s for 5,000 seconds into a 0.1 mm thick polycrystalline tungsten foil with a surface area of 50 x 50 mm$^2$ at room temperature (300 K).

Reviewer: ```suggestion
In this experiment, tritium is implanted at 5 keV and a flux of 1 $\times 10^{19}$ atom/m$_2$/s for 5,000 seconds into a 0.1 mm thick polycrystalline tungsten foil with a surface area of 50 $\times$ 50 mm$^2$ at room temperature (300 K).
```

The background pressure in the implantation chamber is 10$^{-3}$ Pa while the implantation is going on and 10$^{-5}$ Pa the rest of the time.
Following the implantation, the sample is subjected to thermal desorption spectroscopy by heating under vacuum at 50 K/min to 1,273 K and then held at that temperature for several minutes.

> The system is in the structure of [val-2d_schematic]. The implantation chamber (Enclosure 1) has a volume of 0.1 $m^3$ and is evacuated by a turbo-molecular vacuum pump. The implantation chamber is defined for this problem as a enclosure having a preprogrammed temperature of 300 K for 5,000 seconds followed by a ramp to 1,273 K at a ramp rate of 50 K/min. Gas leakage from the ion source is represented by a enclosure with a pressure of $1 \times 10^{-3}$ Pa during implantation followed by $1 \times 10^{-5}$ Pa and flow to the implantation chamber at the vacuum pumping rate. Flow rate from the implantation chamber is taken to be 0.07 m$^3$/s on the basis of the stated pressure in the test chamber during implantation, given that nearly all implanted gas re-emerges during that time. The vacuum pump is represented by a enclosure (Enclosure 2) held at 10-8 Pa.

Reviewer: ```suggestion
The system is in the structure of [val-2d_schematic]. The implantation chamber (Enclosure 1) has a volume of 0.1 $m^3$ and is evacuated by a turbo-molecular vacuum pump. 
The implantation chamber is defined for this problem as a enclosure having a preprogrammed temperature of 300 K for 5,000 seconds followed by a ramp to 1,273 K at a ramp rate of 50 K/min. 
Gas leakage from the ion source is represented by a enclosure with a pressure of $1 \times 10^{-3}$ Pa during implantation followed by $1 \times 10^{-5}$ Pa and flow to the implantation chamber at the vacuum pumping rate. 
Flow rate from the implantation chamber is taken to be 0.07 m$^3$/s on the basis of the stated pressure in the test chamber during implantation, given that nearly all implanted gas re-emerges during that time. The vacuum pump is represented by a enclosure (Enclosure 2) held at 10$^{-8}$ Pa.
```

!media figures/val-2d_schematic.png
        style=width:50%;margin-bottom:2%;margin-left:auto;margin-right:auto
        id=val-2d_schematic
        caption=Schematic of the modeled enclosures for val-2d.

Based on TRIM calculations [!citep](eckstein2013computer,biersack1982stopping), implantation in the sample shows a normal distribution, which has a peak at 4.6 nm below the surface and a characteristic half width of 3 nm. Implantation is activated for 5,000 s and then terminated.

> Three traps are assumed in the sample. Trap concentrations and distributions are considered adjustable parameters while energies were determined by TDS peak temperatures.

Reviewer: ```suggestion
Three trapping site populations are accounted for in the sample. Trap concentrations and distributions are considered adjustable parameters while energies were determined by TDS peak temperatures.
```

D = D_{0} \exp \left( - \frac{E_{D}}{k_b T} \right),
\end{equation}

> where $E_{D}$ is the diffusion activation energy, $k_b$ is the Boltzmann’s constant, $T$ is the temperature, and $D_{0}$ is the maximum diffusivity coefficient. TMAP7 ([!cite](ambrosek2008verification)) assigns two different value, $D_{0,l}$ and $D_{0,r}$, for implantation zone ($x < 15 \times 10 ^ {-9}$ m) and other zone ($x > 15 \times 10 ^ {-9}$ m).

Reviewer: ```suggestion
where $E_{D}$ is the diffusion activation energy, $k_b$ is the Boltzmann’s constant, $T$ is the temperature, and $D_{0}$ is the maximum diffusivity coefficient. TMAP7 ([!cite](ambrosek2008verification)) assigns two different value, $D_{0,l}$ and $D_{0,r}$, for implantation zone ($x < 15 \times 10^{-9}$ m) and other zone ($x > 15 \times 10^{-9}$ m).
```
